cmd/axelard/cmd/vald/broadcaster/broadcast.go

```diff
@@ -45,11 +45,27 @@ func (b *Broadcaster) Broadcast(ctx sdkClient.Context, msgs ...sdk.Msg) (*sdk.Tx
 
 		response, err = Broadcast(ctx, txf, msgs)
 		if err != nil {
-			// reset account and sequence number in case they were the issue
-			b.txFactory = b.txFactory.
-				WithAccountNumber(0).
-				WithSequence(0)
-			return err
+			if !isABCIError(err) {
+				return err
+			}
+
+			// retry on incorrect account sequence
+			if sdkerrors.ErrWrongSequence.Is(err) {
+				// reset account and sequence number in case they were the issue
+				b.txFactory = b.txFactory.
+					WithAccountNumber(0).
+					WithSequence(0)
+				return err
+			}
+
+			// retry on out of gas
+			if sdkerrors.ErrOutOfGas.Is(err) {
+				return err
+			}
+
+			// do not retry on other ABCI error
+			b.logger.Debug(fmt.Sprintf("tx response with error: %s", err))
```

Comment:
what's the rational behind completely suppressing execution errors?

Reply:
for other execution errors, doesn't retry still return the same error?

Reply:
you are literally returning nil, if the error is an `ABCIError` but not wrong sequence or out of gas.Should line 68 be `return err`?